content/articles/automation_lessons.md

```diff
@@ -0,0 +1,110 @@
+title: Automation Lessons
+tags: software engineering, collaboration
+author: cdent
+created: 2025-02-26
+
+```intro
+I initially wrote the following for work colleagues, but sharing more widely
+in case it might apply elsewhere.
+
+A lesson from long ago that has stuck with me and been proven valuable over and
+over:
+
+> When creating software, ensure there are working harnesses for testing,
+> installation, and upgrade first.
+```
+
+Over the years as development technologies have matured the lesson has evolved
+to assert the harnesses must be _automated_ and that the automation _must run
+before changes are merged_.
```

Comment:
```suggestion
to assert the harnesses must be _automated_ and that the _automation must run
before changes are merged_.
```

Struck me as oddly incomplete/unbalanced otherwise?